### PR TITLE
[Environments] Adding configuration flow

### LINKED ISSUE #2668: Add option for a user to 'configure' an environment from the environments page
### Suggested new feature or improvement

In the actions for an environment, add the option to 'configure' an environment.

### Scenario

- User creates a new environment
- Once the environment is created, the user sees the new card on the environments page and wants to take action on that environment
- User should be able to select to 'configure' the environment, which would bring them into the environment configuration flow, with that environment already selected. 

### Additional details

_No response_

### PR DESCRIPTION
**Pull request by huzaifa-d**
## Summary of the pull request
This PR adds the feature to start the configuration flow from the Environments page.
Adds a new button, Setup, which directly takes the UI to the setup flow page.

https://github.com/user-attachments/assets/906ff24c-afd8-4c99-a60c-023d1e5a4eb6

## PR checklist
- [ ] Closes #2668 
- [ ] Tests added/passed
- [ ] Documentation updated

### CODE REVIEW

File: tools/Environments/DevHome.Environments/Strings/en-us/Resources.resw [hunk 1]
```diff
@@ -358,5 +358,8 @@
   <data name="MoreOptionsButton.[using:Microsoft.UI.Xaml.Automation]AutomationProperties.Name" xml:space="preserve">
     <value>More options</value>
     <comment>Name of the button that brings up the "More options" menu</comment>
+  <data name="Operations_ApplyConfiguration" xml:space="preserve">
+    <value>Setup</value>
```
Comment (krschau) [inline tools/Environments/DevHome.Environments/Strings/en-us/Resources.resw:362]: "Set up" as a verb is two words. Or do we mean the noun? (We get this wrong so often and @cinnamon-msft always catches it 🙂)

File: tools/SetupFlow/DevHome.SetupFlow/ViewModels/MainPageViewModel.cs [hunk 2]
```diff
@@ -226,6 +227,25 @@ private void StartSetupForTargetEnvironment(string flowTitle)
             _host.GetService<AppManagementTaskGroup>());
     }
 
+    /// <summary>
+    /// Starts the setup target flow from the environments page.
+    /// </summary>
+    public void StartSetupForTargetEnvironmentWithTelemetry(string flowTitle, string navigationAction, string originPage, ComputeSystemReviewItem item)
```
Comment (krschau) [inline tools/SetupFlow/DevHome.SetupFlow/ViewModels/MainPageViewModel.cs:233]: WithTelemetry? Is there a version that doesn't have telemetry?

File: tools/SetupFlow/DevHome.SetupFlow/ViewModels/SetupFlowViewModel.cs [hunk 3]
```diff
@@ -123,30 +126,52 @@ public async Task StartFileActivationFlowAsync(StorageFile file)
...
+    {
+        Orchestrator.FlowPages = [_mainPageViewModel];
+
+        // This method is only called when the user clicks a button that redirects them to 'Setup' flow in the Environments page.
+        _mainPageViewModel.StartSetupForTargetEnvironmentWithTelemetry(string.Empty, _configurationFlowNavigationParameter, originPage, item);
+    }
+
     public void OnNavigatedTo(NavigationEventArgs args)
     {
         // The setup flow isn't setup to support using the navigation service to navigate to specific
```
Comment (krschau) [inline tools/SetupFlow/DevHome.SetupFlow/ViewModels/SetupFlowViewModel.cs:147]:
```suggestion
        // The setup flow isn't set up to support using the navigation service to navigate to specific
```
Once you start seeing it you can't stop 🙂

File: tools/Environments/DevHome.Environments/ViewModels/ComputeSystemViewModel.cs [hunk 4]
```diff
@@ -73,6 +75,7 @@ public partial class ComputeSystemViewModel : ComputeSystemCardBase, IRecipient<
         IComputeSystem system,
         ComputeSystemProvider provider,
         Func<ComputeSystemCardBase, bool> removalAction,
+        Action<ComputeSystemReviewItem>? configurationAction,
```
[the remark this answers is not in the document]
Comment (huzaifa-d) [reply tools/Environments/DevHome.Environments/ViewModels/ComputeSystemViewModel.cs:78]: Gonna skip this for a later PR that might add more; quite a few examples with 7 or more parameters in the constructor.

File: tools/Environments/DevHome.Environments/Helpers/DataExtractor.cs [hunk 5]
```diff
@@ -151,6 +152,12 @@ public static List<OperationsViewModel> FillLaunchButtonOperations(ComputeSystem
                 _stringResource.GetLocalized("Operations_Terminate"), "\uEE95", computeSystem.TerminateAsync, ComputeSystemOperations.Terminate));
         }
 
+        if (supportedOperations.HasFlag(ComputeSystemOperations.ApplyConfiguration) && configurationCallback is not null)
+        {
+            operations.Add(new OperationsViewModel(
```
[the remark this answers is not in the document]
Comment (huzaifa-d) [reply tools/Environments/DevHome.Environments/Helpers/DataExtractor.cs:157]: Same as above.

File: tools/Environments/DevHome.Environments/ViewModels/OperationsViewModel.cs [hunk 6]
```diff
@@ -5,6 +5,7 @@
 using System.Threading.Tasks;
 using CommunityToolkit.Mvvm.Input;
 using CommunityToolkit.Mvvm.Messaging;
```
[the remark this answers is not in the document]
Comment (huzaifa-d) [reply tools/Environments/DevHome.Environments/ViewModels/OperationsViewModel.cs:7]: Same as above.

File: tools/SetupFlow/DevHome.SetupFlow/ViewModels/MainPageViewModel.cs [hunk 7]
```diff
@@ -226,6 +227,25 @@ private void StartSetupForTargetEnvironment(string flowTitle)
...
+        StartSetupFlowForTaskGroups(
+            flowTitle,
+            "SetupTargetEnvironment",
+            setupTask,
+            _host.GetService<RepoConfigTaskGroup>(),
+            _host.GetService<AppManagementTaskGroup>());
+
+        setupTask.ConfigureTask.SetTargetComputeSystem(item);
+        Orchestrator.GoToNextPage().GetAwaiter().GetResult();
+    }
```
[the remark this answers is not in the document]
Comment (huzaifa-d) [reply tools/SetupFlow/DevHome.SetupFlow/ViewModels/MainPageViewModel.cs:264]: Added.

File: tools/SetupFlow/DevHome.SetupFlow/ViewModels/SetupFlowViewModel.cs [hunk 8]
```diff
@@ -123,30 +126,52 @@ public async Task StartFileActivationFlowAsync(StorageFile file)
...
+            {
+                // We expect that when navigating from anywhere in Dev Home to the create environment page
+                // that the arg.Parameter variable be an object array with the the first value being 'StartCreationFlow',
+                // the second value being the page name that redirection came from for telemetry purposes, and
+                // the third value being the ComputeSystemReviewItem to setup.
+                Cancel();
+                StartSetupFlow(originPage: parameters[1] as string, item: parameters[2] as ComputeSystemReviewItem);
+            }
+        }
+        else
```
[the remark this answers is not in the document]
Comment (huzaifa-d) [reply tools/SetupFlow/DevHome.SetupFlow/ViewModels/SetupFlowViewModel.cs]: Reverted.

File: tools/Environments/DevHome.Environments/Strings/en-us/Resources.resw (hunk 1 above)
[the remark this answers is not in the document]
Comment (huzaifa-d) [reply tools/Environments/DevHome.Environments/Strings/en-us/Resources.resw]: Set up to be consistent with the Setup flow.

File: tools/Environments/DevHome.Environments/ViewModels/ComputeSystemViewModel.cs [hunk 9]
```diff
@@ -133,7 +137,7 @@ private async Task InitializeOperationDataAsync()
             ShouldShowDotOperations = false;
             ShouldShowSplitButton = false;
 
-            RegisterForAllOperationMessages(DataExtractor.FillDotButtonOperations(ComputeSystem, _mainWindow), DataExtractor.FillLaunchButtonOperations(ComputeSystem));
+            RegisterForAllOperationMessages(DataExtractor.FillDotButtonOperations(ComputeSystem, _mainWindow), DataExtractor.FillLaunchButtonOperations(_provider, ComputeSystem, _configurationAction));
```
[the remark this answers is not in the document]
Comment (huzaifa-d) [reply tools/Environments/DevHome.Environments/ViewModels/ComputeSystemViewModel.cs:140]: Same as above.

File: tools/SetupFlow/DevHome.SetupFlow/Models/ConfigureTargetTask.cs [hunk 10]
```diff
@@ -382,6 +383,11 @@ public void RemoveAdaptiveCardPanelFromLoadingUI()
         });
     }
 
+    public void SetTargetComputeSystem(ComputeSystemReviewItem item)
+    {
+         _computeSystemManager.ComputeSystemSetupItem = item;
```
[the remark this answers is not in the document]
Comment (huzaifa-d) [reply tools/SetupFlow/DevHome.SetupFlow/Models/ConfigureTargetTask.cs]: Moved to a background thread.

File: tools/Environments/DevHome.Environments/ViewModels/ComputeSystemViewModel.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (dhoehna) [reply tools/Environments/DevHome.Environments/ViewModels/ComputeSystemViewModel.cs:78]: Adding my 2-cents.  A ctor with many parameters tells me that this class is trying to do too many things.  I wonder if this class can be split?  No need to respond to this comment.

**Review by dhoehna — APPROVED** [review]

**Review by sshilov7 — APPROVED** [review]

**Pull request merged**